add club domain, mapper with test

trigonotis97 opened the pull request:
**동호회 도메인인 Club 도메인, club mapper, test code 추가**

- 특이사항
lombok에서 곧 사라질 기능으로 분류되는 @EqualsAndHashCode 의 of 파라미터 를 대체하기 위해
@EqualsAndHashCode(onlyExplicitlyIncluded = true) 를 사용하였습니다.

onlyExplicitlyIncluded 를 true 로 설정하면 명시적으로 포함된 필드만 해당 메소드(equals(), hashCode())에 사용할수 있습니다.
@EqualsAndHashCode.Include 를 원하는 필드위에 붙여 사용합니다.

api문서 : https://projectlombok.org/api/lombok/EqualsAndHashCode.html#of()

noah0504789 committed:
사용자 계정 - 회원가입 (#12)

* setting build.gradle

* modify build.gradle and setting settings.gradle

* .gitignore

* UserController에 필요한 Form과 Test 코드 (+2 squashed commits)
Squashed commits:
[b42afed] create user domain without JPA
[1d8414d] SpringBootApplication

* using jasypt for application.yml

application.yml with jasypt

* 회원가입 - controller, service, securityConfig

* initialize mybatis for User domain

* AppConfig & ExceptionAdvisor (+1 squashed commit)
Squashed commits:
[b334a07] userMapper & signUpDto unitTest

* modify build.gradle

1. formatting을 적용한 build.gradle

* Delete .diff

PR시 자동생성된 파일.
수정된 .gitignore 에 확장자를 등록하였음으로 제외됩니다.

* formatting to registerUser method

* @Qualifier 주석

부족한 설명을 좀 더 추가하였습니다.

* @Primary 주석

@Primary 어노테이션 설명 추가

noah0504789 committed:
사용자 계정 - 회원가입 (#12)

* setting build.gradle

* modify build.gradle and setting settings.gradle

* .gitignore

* UserController에 필요한 Form과 Test 코드 (+2 squashed commits)
Squashed commits:
[b42afed] create user domain without JPA
[1d8414d] SpringBootApplication

* using jasypt for application.yml

application.yml with jasypt

* 회원가입 - controller, service, securityConfig

* initialize mybatis for User domain

* AppConfig & ExceptionAdvisor (+1 squashed commit)
Squashed commits:
[b334a07] userMapper & signUpDto unitTest

* modify build.gradle

1. formatting을 적용한 build.gradle

* Delete .diff

PR시 자동생성된 파일.
수정된 .gitignore 에 확장자를 등록하였음으로 제외됩니다.

* formatting to registerUser method

* @Qualifier 주석

부족한 설명을 좀 더 추가하였습니다.

* @Primary 주석

@Primary 어노테이션 설명 추가

noah0504789 committed:
사용자 계정 - 회원가입 (#12)

* setting build.gradle

* modify build.gradle and setting settings.gradle

* .gitignore

* UserController에 필요한 Form과 Test 코드 (+2 squashed commits)
Squashed commits:
[b42afed] create user domain without JPA
[1d8414d] SpringBootApplication

* using jasypt for application.yml

application.yml with jasypt

* 회원가입 - controller, service, securityConfig

* initialize mybatis for User domain

* AppConfig & ExceptionAdvisor (+1 squashed commit)
Squashed commits:
[b334a07] userMapper & signUpDto unitTest

* modify build.gradle

1. formatting을 적용한 build.gradle

* Delete .diff

PR시 자동생성된 파일.
수정된 .gitignore 에 확장자를 등록하였음으로 제외됩니다.

* formatting to registerUser method

* @Qualifier 주석

부족한 설명을 좀 더 추가하였습니다.

* @Primary 주석

@Primary 어노테이션 설명 추가

update .gitignore

add git PR files (.diff)

trigonotis97 committed:
user mapper test

trigonotis97 committed:
user mapper test

trigonotis97 committed:
add Club domain, Club Mapper with testcode. add Club Table query.

trigonotis97 committed:
add Club domain, Club Mapper with testcode. add Club Table query.

trigonotis97 committed:
add Club domain, Club Mapper with testcode. add Club Table query.

trigonotis97 committed:
add Club domain, Club Mapper with testcode. add Club Table query.

trigonotis97 committed:
Add newline for end of file

trigonotis97 committed:
Add newline for end of file

trigonotis97 committed:
Add newline for end of file

trigonotis97 committed:
add Club domain, Club Mapper with testcode. add Club Table query.Merge branch 'feature/club-domain-mapper' of https://github.com/trigonotis97/hcs into feature/club-domain-mapper

# Conflicts:
#	src/main/resources/sql/create.sql

trigonotis97 committed:
delete temporary test file and
not necessary blank lines

trigonotis97 committed:
delete temporary test file

trigonotis97 committed:
delete temporary test file and
not necessary blank lines

trigonotis97 committed:
delete temporary test file and
not necessary blank lines

noah0504789 commented:
네 코드 확인하였습니다. 저는 머지해도 좋을 것 같습니다.

f-lab-lyan commented:
천화님, 코드를 보면, 중강중간에 두 줄씩 떨어져 있는 부분이 있는데요. 이 건 코드 포멧팅을 돌려도 여전히 두 줄인가요? 제가 알기로는 한 줄로 줄어드는 걸로 알고 있는데 ... 혹시 확인해 주실 수 있으세요?

f-lab-lyan commented on src/main/resources/mybatis-mapper/ClubMapper.xml:
```diff
@@ -0,0 +1,24 @@
+        values (#{title}, #{description}, #{createdAt}, #{location}, #{category})
+    </insert>
+
+    <delete id="delete" parameterType="String">
```
Delete를 title로 하면, 굳이 id를 별도로 둘 필요가 있을까요? 그리고 별도의 title을 위한 index도 필요하지 않을까요? 위에 selectByTitle때문에라도요. 그리고 이렇게 Table을 만들면, title이 바뀔 경우, 꽤 많은 부분이 바뀌어야 할 것 같은데요. 그 부분에 대해서 신경을 써야할 것 같네요.

그리고 Club이 지금 동호회 자체를 말하는 것 같은데요. 그럼 모든 동아리에서 같은 제목을 하나밖에 못 쓰는 걸로 보여요.

trigonotis97 replied:
- 멘토님께서 지적해주신대로 delete는 id의 존재의미에 맞게 title보단 id로 검색하도록 수정하겠습니다.
- 동호회 이름을 unique 하게 만들었는데, 다시 생각해보니 불필요한것 같습니다. table 생성 쿼리도 같이 수정하겠습니다!
- title이 바뀔경우 많은 부분이 바뀌어야 하는 문제에 대해서는 club 설정 수정서비스 로직을 추가하면서 데이터 수정용 mapper를 별도로 작성하려고 합니다.
- 별도의 title을 위한 index 내용에 대해서는 어떻게 수정해야할지 잘 모르겠습니다... 힌트를 주시면 감사하겠습니다!

trigonotis97 commented:
@f-lab-lyan 빈 두줄인 상태로 자동정렬을 하면 여전히 2줄인데, 1줄로 바꾸거나 아예 빈 줄을 없애고 자동정렬을 하면 1줄로 바뀌더라구요.
인텔리제이 코드 스타일을 수정해서 1줄씩만 떨어지도록 바꾸었습니다. 수정하여 pr에 반영하였습니다.

trigonotis97 committed:
Merge branch 'feature/club-domain-mapper' of https://github.com/trigonotis97/hcs into feature/club-domain-mapper

noah0504789 committed:
reorganize directory and build.gradle for multi-project (re-pullRequest) (#18)

update .gitignore

add git PR files (.diff)

reorganizing directory for multi-project

add Word Space between @Value and explanatory notes

noah0504789 committed:
사용자 계정 - Refactoring (#24)

* Apply @ConfigurationProperties

* rename mpath to mapperLocation

* Return the request error to JSON type

* Specifying the version as an ext variable. (gradleDependencyMgmtPlugin)

* Update @EqualsAndHashCode

* add print() to UserControllerTest

* modify MessageSource

* add spring-boot-actuator

noah0504789 committed:
Update issue templates

trigonotis97 committed:
modify clubmapper - delete, make Club table title un-unique

trigonotis97 committed:
modify clubmapper - delete, make Club table title un-unique

trigonotis97 committed:
modify clubmapper - delete, make Club table title un-unique

trigonotis97 committed:
modify clubmapper - delete, make Club table title un-unique

noah0504789 committed:
사용자 계정 - Validation Test (#28)

* create test

* create @EnableMockMvc for adding CharacterEncodingFilter

* using @EnableMockMvc

* create TestUserControllerWithoutValid

* create test for not using @Valid

* modify files during merge

noah0504789 committed:
apply SSL (#29)

* apply SSL

trigonotis97 committed:
Merge branch 'feature/club-domain-mapper' of https://github.com/trigonotis97/hcs into feature/club-domain-mapper

trigonotis97 committed:
resolve conflict

trigonotis97 committed:
Merge branch 'feature/club-domain-mapper' of https://github.com/trigonotis97/hcs into feature/club-domain-mapper
